tiny_dnn/core/kernels/fully_connected_op_cblas.h

```diff
@@ -0,0 +1,48 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/core/params/fully_params.h"
+
+#ifdef CNN_USE_CBLAS
+extern "C" {
+#include <cblas.h>
+}
+#endif
+
+namespace tiny_dnn {
+namespace kernels {
+
+inline void fully_connected_op_cblas(const tensor_t &in_data,
+                                     const vec_t &W,
+                                     const vec_t &bias,
+                                     tensor_t &out_data,
+                                     const core::fully_params &params,
+                                     const bool layer_parallelize) {
+#ifdef CNN_USE_CBLAS
+  size_t out_size = params.out_size_;
+  size_t in_size  = params.in_size_;
+  float_t alpha   = 1;
+  float_t beta    = 1;
+  float_t *input  = reinterpret_cast<float_t *>(in_data[0].data());
+  float_t *weight = reinterpret_cast<float_t *>(W.data());
+  float_t *output = out_data[0].data();
+  memcpy(output, bias.data(), sizeof(float_t) * out_size);
+#ifdef CNN_USE_DOUBLE
+  cblas_dgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, 1, out_size, in_size,
+              alpha, input, in_size, weight, out_size, beta, output, out_size);
+#else
+  cblas_sgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, 1, out_size, in_size,
+              alpha, input, in_size, weight, out_size, beta, output, out_size);
+#endif
+
+#endif  // CNN_USE_CBLAS
```

Comment:
if `CNN_USE_CBLAS` isn't defined but `fully_connected_op_cblas` is called, throwing `nn_error` exception might be preferrable. But... how to deal with the situation isn't consistent throughout the library. So I guess it's not mandatory.  :confused:

By the way, could you kindly tell me which library I should use to test this layer? OpenBLAS?